packages/frontend/components/scaffold-eth/Contract/utilsContract.tsx

```diff
@@ -0,0 +1,165 @@
+import { FunctionFragment } from "ethers/lib/utils";
+import ContractData from "~~/contracts/hardhat_contracts.json";
+import { Contract, utils } from "ethers";
+import DisplayVariable from "~~/components/scaffold-eth/Contract/DisplayVariables";
+import { ReadOnlyFunctionForm } from "./ReadOnlyFunctionForm";
+import { WriteOnlyFunctionForm } from "./WriteOnlyFunctionForm";
+
+type GeneratedContractType = {
+  address: string;
+  abi: any[];
+};
+
+// TODO Add checks if the contract is not present at `contracts` directory
+/**
+ * @param chainId - deployed contract chainId
+ * @param contractName - name of deployed contract
+ * @returns {GeneratedContractType} object containing contract address and abi
+ */
+const getDeployedContract = (
+  chainId: string | undefined,
+  contractName: string | undefined | null,
+): GeneratedContractType | undefined => {
+  if (!chainId || !contractName) {
+    return { address: "", abi: [] };
+  }
+
+  const contractsAtChain = ContractData[chainId as keyof typeof ContractData];
+  const contractsData = contractsAtChain[0]?.contracts;
+
+  return contractsData[contractName as keyof typeof contractsData];
+};
```

Comment:
@technophile-04 Regarding "find a better way contract from contracts directory, currently, it's a hardcoded"

I think it's good enough as it's rn. This is how `hardhat_contracts.json` looks when you deploy a second contract in 31337:

![image](https://user-images.githubusercontent.com/2486142/201068690-88f19599-3a19-4318-b52f-56e87b55d00e.png)

I think `contractsAtChain[0]` is always going to return the right thing, since we are already inside the 31337 array.

I tweaked the name of the functions and fixed some issues with types.